### PR TITLE
fixed calculate ranges

### PR DESCRIPTION
**Pull request by cseed**
rounding could cause last partition to be mis-sized
also, normalize values before shuffle

### CODE REVIEW

File: src/main/scala/is/hail/rvd/OrderedRVD.scala [hunk 1]
```diff
@@ -540,27 +542,28 @@ object OrderedRVD {
...
-      if (keys.length <= nPartitions + 1)
-        keys
-      else {
-        val k = keys.length / nPartitions
-        assert(k > 0)
-        Array.tabulate(nPartitions)(i => keys(i * k)) :+ max
-      }
+    if (pkOrd.compare(ab.last, max) != 0)
+      ab += max
+    val partitionMaxes = ab.result()
```
Comment (jbloom22) [inline src/main/scala/is/hail/rvd/OrderedRVD.scala]: This contains the minimum too, more consistent to call it `rangeBounds` as used for the parameter in `OrderedRVDPartitioner.makeRangeBoundIntervals`?

Comment (cseed) [reply src/main/scala/is/hail/rvd/OrderedRVD.scala]: Good catch, I had meant to rename that `partitionEdges` (following the `hist`).  Done.

**Comment by cseed** [comment]
No, it's actually the edges: the min and max of the first, and then the max of all the rest.

**Comment by jbloom22** [comment]
I know, twas a joke!

**Comment by jbloom22** [comment]
(note the ;) and approval)

**Pull request merged**